File: platforms/python/examples/core/pass/add_depth_target.py
from fragmentcolor import Renderer, Pass, Shader, Mesh

renderer = Renderer()
target = renderer.create_texture_target([64, 64])

# One depth attachment shared across the 3D-content pass.
depth = renderer.create_depth_texture([64, 64])

mesh = Mesh()
mesh.add_vertex([0.0, 0.0, 0.0])
mesh.add_vertex([1.0, 0.0, 0.0])
mesh.add_vertex([0.0, 1.0, 0.0])
mesh.add_vertex([1.0, 1.0, 0.0])
shader = Shader.from_mesh(mesh)
rpass = Pass("blobs"); rpass.add_shader(shader)

# Depth-test on — closer fragments win, the rpass writes to the depth
# buffer so subsequent draws within the same rpass see the depth.
rpass.add_depth_target(depth)

renderer.render(rpass, target)

File: platforms/swift/examples/core/pass/add_depth_target.swift
import FragmentColor

let renderer = Renderer()
let target = try await renderer.createTextureTarget([64, 64])

// One depth attachment shared across the 3D-content pass.
let depth = try await renderer.createDepthTexture([64, 64])

let mesh = Mesh()
try mesh.addVertex([0.0, 0.0, 0.0])
try mesh.addVertex([1.0, 0.0, 0.0])
try mesh.addVertex([0.0, 1.0, 0.0])
try mesh.addVertex([1.0, 1.0, 0.0])
let shader = Shader.fromMesh(mesh)
let pass = Pass("blobs"); pass.addShader(shader)

// Depth-test on — closer fragments win, the pass writes to the depth
// buffer so subsequent draws within the same pass see the depth.
try pass.addDepthTarget(depth)

try renderer.render(pass, target)